linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -913,20 +920,54 @@ private final class ClassDefChecker(classDef: ClassDef,
     if (!declaredLabelNamesPerMethod.add(label.name))
       reportError(i"Duplicate label named ${label.name}.")
   }
+
+  private def checkLinkTimeCond(cond: LinkTimeTree)(
+      implicit ctx: ErrorContext): Unit = {
+    if (cond.tpe != BooleanType)
+      reportError(i"Link time condition must be typed as boolean, but ${cond.tpe} is found.")
+    checkLinkTimeTree(cond)
+  }
+
+  private def checkLinkTimeTree(cond: LinkTimeTree)(
+      implicit ctx: ErrorContext): Unit = {
```

Comment:
This one should not take an implicit `ErrorContext`. Instead, it should build its own local `ErrorContext` from the `LinkTimeTree` argument. Like how `def checkTree` does.